common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala

```diff
@@ -2917,12 +2917,12 @@ object CelebornConf extends Logging {
   val WORKER_GRACEFUL_SHUTDOWN_RECOVER_DB_BACKEND: ConfigEntry[String] =
     buildConf("celeborn.worker.graceful.shutdown.recoverDbBackend")
       .categories("worker")
-      .doc("Specifies a disk-based store used in local db. LEVELDB or ROCKSDB.")
+      .doc("Specifies a disk-based store used in local db. ROCKSDB or LEVELDB (deprecated).")
       .version("0.4.0")
       .stringConf
       .transform(_.toUpperCase(Locale.ROOT))
       .checkValues(Set("LEVELDB", "ROCKSDB"))
-      .createWithDefault("LEVELDB")
+      .createWithDefault("ROCKSDB")
```

Comment:
Wont this not cause issues given this is a change in patch version ?

Reply:
why should we make such changes on a patched version?

Reply:
@mridulm, I have cherry pick #2369, #2372 to branch-0.4. The failure of CI is caused by these pull request, not caused by the default value change.
BTW, @onebox-li, could you rebase branch-0.4 again? I have adjusted the order of cherry pick.

Reply:
@pan3793, add a note in the upgrade guide to explain the situation. Meanwhile, the core reason is that leveldb cannot run on arm Linux.

Reply:
To clarify my query - if the default value is changing due to this PR - that will break existing 0.4 deployments: for example when they are not configuring this config, and now end up with a change in `recoverDbBackend`

Reply:
@mridulm, I understood your point. I have mentioned above that we would add a note to explain this behavior. The users could add a config that changes to LEVELDB.

Reply:
If 0.4 is not working on arm currently - wouldn't it not be better for arm deployments to explicitly set it to ROCKSDB ? (in other words, not to change the default ?)

To put it differently, my main concern is failure of existing deployments ... in our case, for example, we do not set this config (though we are still in testing phase) - and we would typically expect a patch to work reasonably as is with existing config (between minor or major versions, there will be a lot more scrutiny).

Reply:
@mridulm, you are right. I have removed this cherry pick from branch-0.4. I would like to resolve [CELEBORN-1380](https://issues.apache.org/jira/browse/CELEBORN-1380) for arm.

Reply:
Please correct fixed versions for CELEBORN-1380, it should not go 0.4.1

Reply:
cc @FMX, WDYT?